sdks/go/pkg/beam/io/spannerio/read_batch_integration_test.go

```diff
@@ -0,0 +1,126 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package spannerio
+
+import (
+	"cloud.google.com/go/spanner"
+	database "cloud.google.com/go/spanner/admin/database/apiv1"
+	"context"
+	"flag"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/testing/passert"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/testing/ptest"
+	adminpb "google.golang.org/genproto/googleapis/spanner/admin/database/v1"
+	"reflect"
+	"testing"
+)
+
+var (
+	integrationTests = flag.Bool("int", false, "")
+)
+
+func TestExampleQueryBatch(t *testing.T) {
+	if !*integrationTests {
+		t.Skip("Not running in integration test mode.")
+	}
+
+	p := beam.NewPipeline()
+	s := p.Root()
+
+	db := "projects/test-project/instances/test-instance/databases/test-database"
+
+	srv, srvCleanup := newServer(t)
+	defer srvCleanup()
+
+	client, admin, cleanup, err := createFakeClient(srv.Addr, db)
+	if err != nil {
+		t.Fatalf("Unable to create fake client: %v", err)
+	}
+	defer cleanup()
+
+	populateSpanner(context.Background(), admin, db, client)
+
+	rows := QueryBatch(s, db, "SELECT * FROM TEST", reflect.TypeOf(TestDto{}))
+
+	ptest.RunAndValidate(t, p)
+	passert.Count(s, rows, "Should have 4 rows", 4)
+}
+
+func populateSpanner(ctx context.Context, admin *database.DatabaseAdminClient, db string, client *spanner.Client) error {
```

Comment:
Since this is part of test set up it should take in a `*testing.T` and call `t.Helper()`. All errors should then be logged with the `t.Error` or `t.Fatal` etc, assuming they block execution of the test. 

As it stands, all the errors are ignored, which would make debugging difficult.

See https://google.github.io/styleguide/go/decisions#test-helpers